src/ui/public/courier/__tests__/saved_object.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Tests functionality in ui/public/courier/saved_object/saved_object.js
+ */
+
+import ngMock from 'ng_mock';
+import expect from 'expect.js';
+import sinon from 'auto-release-sinon';
+
+import BluebirdPromise from 'bluebird';
+import SavedObjectFactory from '../saved_object/saved_object';
+import { stubMapper } from 'test_utils/stub_mapper';
+import IndexPatternFactory from 'ui/index_patterns/_index_pattern';
+
+describe('Saved Object', function () {
+  require('test_utils/no_digest_promises').activateForSuite();
+
+  let savedObjectFactory;
+  let IndexPattern;
+  let esStub;
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject(function ($injector, Private) {
+    savedObjectFactory = Private(SavedObjectFactory);
+    IndexPattern = Private(IndexPatternFactory);
+    esStub = $injector.get('es');
+
+    mockEsService();
+    stubMapper(Private);
+  }));
+
+  /**
+   * Some default es stubbing to avoid timeouts and allow a default type of 'dashboard'.
+   */
+  function mockEsService() {
+    // Allows the type 'dashboard' to be used.
+    // Unfortunately we need to use bluebird here instead of native promises because there is
+    // a call to finally.
+    sinon.stub(esStub.indices, 'getFieldMapping').returns(BluebirdPromise.resolve({
+      '.kibana' : {
+        'mappings': {
+          'dashboard': {}
+        }
+      }
+    }));
+
+    // Necessary to avoid a timeout condition.
+    sinon.stub(esStub.indices, 'putMapping').returns(BluebirdPromise.resolve());
+  };
+
+  /**
+   * Stubs some of the es retrieval calls so it returns the given response.
+   * @param {Object} mockDocResponse
+   */
+  function stubESResponse(mockDocResponse) {
+    sinon.stub(esStub, 'mget').returns(BluebirdPromise.resolve({ docs: [mockDocResponse] }));
+    sinon.stub(esStub, 'index').returns(BluebirdPromise.resolve(mockDocResponse));
+  }
+
+  /**
+   * Creates a new saved object with the given configuration. Does not call init.
+   * @param {Object} config
+   * @returns {SavedObject} an instance of SavedObject
+   */
+  function createSavedObject(config = {}) {
+    let savedObject = {};
+    savedObjectFactory.call(savedObject, config);
+    return savedObject;
+  }
+
+  describe ('config', function () {
+
+    it('afterESResp is called', function () {
+      let afterESRespCallback = sinon.spy();
```

Comment:
Throughout the tests, there's an inconsistent usage of `const` and `let`. There appears to be a pattern that generally `let` is used when an object will be 'mutated'; however, that's not the difference between `const`/`let` in javascript. `let` allows a reference to be reassigned per https://github.com/airbnb/javascript#references--prefer-const